src/strcasestr.h

```diff
@@ -0,0 +1,45 @@
+
+/*
+ * Polyfill for strcasestr function
```

Comment:
It appears that the state of case-insensitive string comparisons is not good, even in C++20.  :disappointed:

Reply:
Actually, this should be part of C89 and C99?

Reply:
We only use `strcasestr` once in `MapPutErrorResponse` -- maybe it is better to just lowercase the entire string or use the correct XML parsing functions to avoid this polyfill?

Reply:
This PR is scoped to porting s3fs-fuse to Windows; I think improving existing code should be another issue 🤔

Reply:
This isn't an improvement; it is working around win32 lack of support.  I opened #1858 to address the `strcasestr` comment but expect you to address the `strptime` comment.  Importing a bunch of random code imposes a maintenance cost and it is your responsibility to minimize this.

Reply:
> Importing a bunch of random code imposes a maintenance cost and it is your responsibility to minimize this.

I agree with this.
So should I merge #1858 and delete this polyfill after you merged the PR?

Reply:
Finally I merged #1858 into HEAD of this PR. Merge this PR after #1858 was merged.